File: api/src/main/java/fr/huiitre/tools/modules/riot/valorant/application/usecase/GetValorantBundleByAssetIdUseCase.java
package fr.huiitre.tools.modules.riot.valorant.application.usecase;

import java.util.Optional;
import java.util.UUID;

import org.springframework.stereotype.Service;

import fr.huiitre.tools.modules.core.module.domain.ModuleCode;
import fr.huiitre.tools.modules.core.role.domain.RoleCode;
import fr.huiitre.tools.modules.core.security.application.usecase.SecuredUseCase;
import fr.huiitre.tools.modules.riot.valorant.application.ports.ValorantBundleRepository;
import fr.huiitre.tools.modules.riot.valorant.application.view.ValorantBundleView;

@Service
public class GetValorantBundleByAssetIdUseCase implements SecuredUseCase {

    private final ValorantBundleRepository bundleRepository;

    public GetValorantBundleByAssetIdUseCase(ValorantBundleRepository bundleRepository) {
        this.bundleRepository = bundleRepository;
    }

    @Override
    public Optional<ModuleCode> requiredModule() {
        return Optional.of(ModuleCode.RIOT);
    }

    @Override
    public RoleCode requiredRole() {
        return RoleCode.READ_ONLY;
    }

    public ValorantBundleView execute(UUID assetId) {
        return bundleRepository.findByAssetId(assetId)
                .orElseThrow(() -> new IllegalArgumentException("Bundle not found: " + assetId));
    }
}

File: api/Modules/Riot/Valorant/Application/Catalog/Usecases/GetValorantBundleByAssetIdUseCase.cs
using Tools.Api.Modules.Core.Common.Application.Exceptions;
using Tools.Api.Modules.Core.Security.Application.Services;
using Tools.Api.Modules.Core.Security.Application.Usecases;
using Tools.Api.Modules.Core.Security.Domain;
using Tools.Api.Modules.Riot.Valorant.Application.Catalog.Ports;
using Tools.Api.Modules.Riot.Valorant.Application.Catalog.Views;

namespace Tools.Api.Modules.Riot.Valorant.Application.Catalog.Usecases;

public sealed class GetValorantBundleByAssetIdUseCase(
    UseCaseAuthorizer authorizer,
    IValorantBundleRepository valorantBundleRepository
) : SecuredUseCase(authorizer)
{
    protected override ModuleCode? RequiredModule => ModuleCode.Riot;
    protected override RoleCode RequiredRole => RoleCode.ReadOnly;

    public async Task<ValorantBundleView> Execute(Guid assetId)
    {
        var bundle = await valorantBundleRepository.FindByAssetId(assetId);

        return bundle ?? throw AppException.NotFound(
            "VALORANT_BUNDLE_NOT_FOUND",
            $"Le pack {assetId} est introuvable.");
    }
}